modules/post/multi/gather/jenkins_gather.rb

```diff
@@ -312,45 +313,59 @@ def get_key_material(home, platform)
         vprint_status("File hudson.util.Secret saved to #{loot_path}")
       end
     else
-      print_error 'Cannot read master.key or hudson.util.Secret...'
-      print_error 'Encrypted strings will not be able to be decrypted...'
+      print_error "Cannot read master.key or hudson.util.Secret..."
+      print_error "Encrypted strings will not be able to be decrypted..."
       return
     end
   end
-
   def find_home(platform)
-    print_status('Searching for Jenkins directory... This could take some time...')
-    case platform
-    when 'windows'
-      case session.type
-      when 'meterpreter'
-        home = session.fs.file.search(nil, 'secret.key.not-so-secret')[0]['path']
-      else
-        home = cmd_exec('cmd.exe', "/c dir /b /s c:\*secret.key.not-so-secret", timeout = 120).split('\\')[0..-2].join('\\').strip
+    if datastore['JENKINS_HOME'] && exist?(datastore['JENKINS_HOME'] + '/secret.key.not-so-secret')
+      home = datastore['JENKINS_HOME']
+    else
```

Comment:
Rather than wrap the entire method in this conditional statement, you could instead do:

```ruby
    if datastore['JENKINS_HOME'] && exist?(datastore['JENKINS_HOME'] + '/secret.key.not-so-secret')
      return datastore['JENKINS_HOME']
    end
```

Better yet, as this relies on a user-specified `datastore` option, and will fall back to "finding" the home directory anyway, this condition could be performed before calling `find_home`. Unfortunately, going down that route reveals a lot more that could be refactored and cleaned up.